test/school-service.test.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+var chai = require('chai');
+var assert = chai.assert;
+var _ = require('lodash');
+var SchoolServiceFactory = require('../src/school-service');
+var RegistrationFormFactory = require('../src/registration-form');
+var UuidFactory = require('../src/uuid-generator');
+
+describe('Testing school-service', function () {
+
+    describe('for valid creation', function () {
+
+        it('should return a school service', function () {
+            var schoolService = SchoolServiceFactory.create();
+
+            assert.isNotNull(schoolService);
+        });
+    });
+
+    describe('create new school', function () {
+
+        var schoolRepositoryStub = {
+            schools: [],
+
+            save: function (school) {
+                this.schools.push(school);
+            },
+
+            getLastSavedSchool: function () {
+                return this.schools[0];
+            }
+        };
+
+        var schoolService = SchoolServiceFactory.create({schoolRepository: schoolRepositoryStub});
+
+        before(function () {
+            var schoolInputData = {
+                name: 'TDD School',
+                number: 5
+            };
+
+            schoolService.createSchool(schoolInputData);
+        });
+
+        it('should save new school', function () {
+            assert.equal(schoolRepositoryStub.getLastSavedSchool().number, 5);
+        });
+    });
+
+
+    describe('#getSchool', function () {
+
+        var schoolRepositoryStub = {
+            schools: [],
+
+            save: function (school) {
+                this.schools.push(school);
+            },
+
+            getLastSavedSchool: function () {
+                return this.schools[0];
+            }
+        };
+
+        var schoolService = SchoolServiceFactory.create({schoolRepository: schoolRepositoryStub});
+        var schoolInputData;
+        before(function () {
+            schoolInputData = {
+                name: 'TDD School',
+                number: 5
+            };
+
+            schoolService.createSchool(schoolInputData);
+        });
+
+        it('should  get the school', function () {
+            assert.equal(schoolService.getSchoolByNumber({number: 5}), schoolRepositoryStub.getLastSavedSchool());
+        });
+    });
+
+    describe('#register', function () {
+
+        var schoolRepositoryStub = {
+            schools: [],
+
+            save: function (school) {
+                this.schools.push(school);
+            },
+
+            getLastSavedSchool: function () {
+                return this.schools[0];
+            }
+        };
+
+        var schoolService = SchoolServiceFactory.create({schoolRepository: schoolRepositoryStub});
+        var schoolInputData;
+
+        var form = {
+            userId: UuidFactory.create({id: 1}),
```

Comment:
when you stub, then stub as close as possible to the real implementation (types)